### ISSUE TITLE
Document setting up IDE integration with BlocksDS

### ISSUE THREAD

**Issue by asiekierka** [blocksds/sdk#231]
From a certain forum:

> The problem is that outside of setup there is no documentation/tutorial at all for integrating BlocksDS support into an IDE, whether it its VS Code, the full fat VS or any other C/C++ IDE available out there. Has anyone tried so far to include BlocksDS functions inside an IDE and compile a program with it in a terminal? I am sorry if this topic exists already.

Wonderful solves this by providing some tools and some patches on top of GCC to generate `compile_commands.json` files that most IDEs can ingest, however this will further deepen BlocksDS's dependency on Wonderful. There are probably other options.

**Comment by GalaxyShard** [comment]
As a quick and dirty solution, it may be worth documenting clangd's compile_flags.txt which can specify options (defines, include paths, target...) across an entire project. It's quite easy to set up and I use it myself in most C and C++ projects I work with--although I don't believe it would work for arm7+arm9 projects given that the options are project-wide. I'm not sure what behavior clangd follows if there are different compile_flags.txt in subdirectories, but that could be a possible solution.

There's also the possibility of making a tool that generates compile_commands.json by traversing the specified source directories and creating entries with the proper arm7/arm9 flags, although something like [bear](https://github.com/rizsotto/Bear) could just be used instead (I've found it to be a bit finicky though).

**Comment by asiekierka** [comment]
Personally, I just went for implementing `-MJ` in GCC.

**Comment by asiekierka** [comment]
> The default makefiles aren't that useful, anyway

I disagree. Users who need IDE integration most are the ones who are just starting out with BlocksDS. Those users are going to be using your tutorial (which onboards them onto using `make`) and your examples (which, generally, use `make`). As such, I think adding some kind of IDE hinting facility (`compile_commands.json` being the most supported standard, as well as something Wonderful's tooling makes fairly easy to add; but `compile_flags.txt` is also a valid option for clangd-based IDEs) would be beneficial.

**Comment by AntonioND** [comment]
https://codeberg.org/blocksds/sdk/issues/231
